apps/admin-page/src/app/components/create-discount-dialog/create-discount-dialog.component.ts

```diff
@@ -0,0 +1,72 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { Component } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { MatDialog } from '@angular/material/dialog';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { DiscountType, IDiscount } from '@interfaces';
+import { DiscountsService } from '../../services/discounts/discounts.service';
+
+@Component({
+  selector: 'webstore-create-discount-dialog',
+  templateUrl: './create-discount-dialog.component.html',
+  styleUrls: ['./create-discount-dialog.component.css'],
+})
+export class CreateDiscountDialogComponent {
+  createDiscountForm: FormGroup;
+  checked = true;
+
+  /**
+   * Initialise the create password form.
+   *
+   * @param discountService
+   * @param snackbar
+   * @param dialog
+   */
+  constructor(private discountService: DiscountsService, private snackbar: MatSnackBar, private dialog: MatDialog) {
+    this.createDiscountForm = new FormGroup({
+      discountCode: new FormControl('', [Validators.maxLength(50), Validators.minLength(3), Validators.required]),
+      startsAt: new FormControl(''),
+      expiresAt: new FormControl('', [Validators.required]),
```

Comment:
Expires at is not required and can be null in the discount

Reply:
I want it to be required so an end date is set.